storybookjs/storybook
# Addon-docs: Change the way PropType used by ArgTable are generated for Flow types

Issue: N/A
Requires: #12588 

## What I did

I changed the way Flow types were converted into `PropType` because ArgsTable expects a string for `summary` and `details`, but Flow types passed to the converter will contain rich type information that is often lost when just using the `raw` value (as was done previously).

The way the generation is done, all the previous tests that just used `raw` values are still passing. Additionally, I added a bunch of tests around more specific types of generation.

It might also make sense to have a discussion around whether `ArgsTable` should receive the rich type information produced by react-docgen instead of only operating on string values. I think that would resolve issues like #12587 in a cleaner manner.

## How to test

- Is this testable with Jest or Chromatic screenshots? Yep, added a bunch of jest snapshots.
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.
